### PR TITLE
feature(whl):add trex example

### PR DESCRIPTION
**Pull request by kxzxvbk**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/classic_control/cartpole/config/cartpole_trex_dqn_config.py [hunk 1]
```diff
@@ -20,9 +20,11 @@
         checkpoint_step=100,
         learning_rate=1e-5,
         update_per_collect=1,
+        num_trajs=6,
```
Comment (Will-Nie) [inline dizoo/classic_control/cartpole/config/cartpole_trex_dqn_config.py:23]: why do we assign num_trajs=6?

File: ding/example/trex.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        task.use(eps_greedy_handler(cfg))
+        task.use(StepCollector(cfg, policy.collect_mode, collector_env))
+        task.use(data_pusher(cfg, buffer_))
+        task.use(OffPolicyLearner(cfg, policy.learn_mode, buffer_, reward_model))
+        task.use(CkptSaver(cfg, policy, train_freq=100))
+        task.run()
+
+
+if __name__ == "__main__":
+    main()
```
Comment (Will-Nie) [inline ding/example/trex.py:59]: do not forget to solve the unittest issue

File: ding/example/trex.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        )
+
+        evaluator_env = BaseEnvManagerV2(
+            env_fn=[lambda: DingEnvWrapper(gym.make("CartPole-v0")) for _ in range(cfg.env.evaluator_env_num)],
+            cfg=cfg.env.manager
+        )
+
+        set_pkg_seed(cfg.seed, use_cuda=cfg.policy.cuda)
+        model = DQN(**cfg.policy.model)
+        buffer_ = DequeBuffer(size=cfg.policy.other.replay_buffer.replay_buffer_size)
```
Comment (Will-Nie) [inline ding/example/trex.py:42]: just curious, why do we add '_' after 'buffer'. Is there any specific meaning for this as if you add '_' as prefix

File: ding/example/trex.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+from ding.framework.context import OnlineRLContext
+from ding.framework.middleware import OffPolicyLearner, StepCollector, interaction_evaluator, \
+    eps_greedy_handler, CkptSaver, eps_greedy_masker, sqil_data_pusher, data_pusher
+from ding.utils import set_pkg_seed
+from dizoo.classic_control.cartpole.config.cartpole_trex_dqn_config import main_config, create_config
+from ding.entry import trex_collecting_data
+from ding.reward_model import create_reward_model
+import copy
+import easydict
+import os
```
Comment (Will-Nie) [inline ding/example/trex.py:20]: Suggestion: put these import module before import ding to make it look better
(you can simply follow this import order:
some official modules --> torch modules --> ding modules --> relative import )

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**